passes/cmds/select.cc

```diff
@@ -896,6 +896,28 @@ static void select_stmt(RTLIL::Design *design, std::string arg)
 	select_filter_active_mod(design, work_stack.back());
 }
 
+static void print_selection_for_assert(RTLIL::Design *design, RTLIL::Selection *sel)
+{
+	fprintf(stderr, "Selection contains:\n");
+	for (auto mod_it : design->modules_)
+	{
+		if (sel->selected_module(mod_it.first)) {
+			for (auto &it : mod_it.second->wires_)
+				if (sel->selected_member(mod_it.first, it.first))
+					fprintf(stderr, "%s/%s\n", id2cstr(mod_it.first), id2cstr(it.first));
+			for (auto &it : mod_it.second->memories)
+				if (sel->selected_member(mod_it.first, it.first))
+					fprintf(stderr, "%s/%s\n", id2cstr(mod_it.first), id2cstr(it.first));
+			for (auto &it : mod_it.second->cells_)
+				if (sel->selected_member(mod_it.first, it.first))
+					fprintf(stderr, "%s/%s\n", id2cstr(mod_it.first), id2cstr(it.first));
+			for (auto &it : mod_it.second->processes)
+				if (sel->selected_member(mod_it.first, it.first))
+					fprintf(stderr, "%s/%s\n", id2cstr(mod_it.first), id2cstr(it.first));
```

Comment:
This shouldn't simply print to stderr.

These `fprintf(stderr, ...)` should either be replaced by `log()` or `log_warning()`, or `print_selection_for_assert` should be replaced by a function that returns a multi-line string that can be included in the error message printed with `log_error` in the callers below.